haystack_experimental/dataclasses/chat_message.py

```diff
@@ -190,9 +190,6 @@ def from_assistant(
         :param meta: Additional metadata associated with the message.
         :returns: A new ChatMessage instance.
         """
-        if not text and not tool_calls:
-            raise ValueError("At least one of `text` or `tool_calls` must be provided.")
```

Comment:
Under what circumstances can we expect empty messages from the LLM? And what makes them non-exceptional?

Reply:
we have empty assistant messages in case of content filtering: https://platform.openai.com/docs/api-reference/chat/object#chat/object-choices (see `choices > finish_reason`)
we handle this circumstance here: https://github.com/deepset-ai/haystack/blob/c4be8370c624c9217f34e996abd7f1d147c17c2d/haystack/components/generators/chat/openai.py#L365